pkgs/applications/science/computer-architecture/timeloop/default.nix

```
@@ -0,0 +1,97 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, scons
+, libconfig
+, boost
+, libyaml
+, libyamlcpp
+, ncurses
+, gpm
+, enableAccelergy ? true
+, enableISL ? false
+, accelergy
+}:
+
+stdenv.mkDerivation rec {
+  pname = "timeloop";
+  version = "unstable-2022-10-28";
+
+  src = fetchFromGitHub {
+    owner = "NVlabs";
+    repo = "timeloop";
+    rev = "83088f2b9b01581685cfe856dd9d84e46575cfdb";
+    hash = "sha256-X/5lvOYSPJTKcdgMEKVQbd34n4INf7P+lTl98yv82fs=";
+  };
+
+  nativeBuildInputs = [ scons ];
+
+  buildInputs = [
+    libconfig
+    boost
+    libyaml
+    libyamlcpp
+    ncurses
+    accelergy
+   ] ++ lib.optionals stdenv.isLinux [ gpm ];
+
+  preConfigure = ''
+     cp -r ./pat-public/src/pat ./src/pat
+   '';
+
+  enableParallelBuilding = true;
+
+  #link-time optimization fails on darwin
+  #see https://github.com/NixOS/nixpkgs/issues/19098
+  NIX_CFLAGS_COMPILE = lib.optional stdenv.isDarwin "-fno-lto";
+
+  patchPhase = ''
+    #use nix ar/ranlib
+    substituteInPlace ./SConstruct --replace "env.Replace(AR = \"gcc-ar\")" ""
+    substituteInPlace ./SConstruct --replace "env.Replace(RANLIB = \"gcc-ranlib\")" ""
+    '' + lib.optionalString stdenv.isDarwin ''
+    #prevent clang from dying on errors that gcc is fine with
+    substituteInPlace ./src/SConscript --replace "-Werror" "-Wno-inconsistent-missing-override"
+
+    #disable LTO on macos
+    substituteInPlace ./src/SConscript --replace ", '-flto'" ""
```

```suggestion
    # use nix ar/ranlib
    substituteInPlace ./SConstruct \
      --replace "env.Replace(AR = \"gcc-ar\")" ""
      --replace "env.Replace(RANLIB = \"gcc-ranlib\")" ""
    '' + lib.optionalString stdenv.isDarwin ''
    # prevent clang from dying on errors that gcc is fine with
    substituteInPlace ./src/SConscript --replace "-Werror" "-Wno-inconsistent-missing-override"

    # disable LTO on macos
    substituteInPlace ./src/SConscript --replace ", '-flto'" ""
```
Please use a space after the hashtag everywhere